### PR TITLE
[CAMEL-14245] camel-main - allow to enable/disable auto confioguiration per component/language/data-format

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

File: core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java [hunk 1]
```diff
@@ -1082,4 +1050,69 @@ public int hashCode() {
...
+            //
+            //   1. ${keyPrefix}.enabled = true|false
+            //   2. ${keyPrefix}.${name}.enabled = true|false
+            //
+            // The option [2] has the higher priority so as example:
+            //
+            //   camel.component.enabled = false
+            //   camel.component.seda.enabled = true
+            //
+            // enables auto configuration of the seda component only
```
Comment (davsclaus) [inline core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java]: Seems code comments is talking about seda

File: core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java [hunk 2]
```diff
@@ -1082,4 +1050,69 @@ public int hashCode() {
...
+            PropertyOptionKey pok = new PropertyOptionKey(key, target, prefix);
+            Map<String, Object> values = properties.computeIfAbsent(pok, k -> new LinkedHashMap<>());
+
+            // we ignore case for property keys (so we should store them in canonical style
+            values.put(optionKey(option), value);
+        }
+    }
+
+    public static boolean isServiceEnabled(String prefix, String name, Properties properties) {
+        ObjectHelper.notNull(prefix, "prefix");
```
Comment (davsclaus) [inline core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java]: copy paste error here in these 3 lines

File: core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java (hunk 1 above)
Comment (lburgazzoli) [reply core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java]: This is expected as it is an example, should I change it ?

File: core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java (hunk 2 above)
Comment (lburgazzoli) [reply core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java]: done

File: core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java (hunk 1 above)
Comment (davsclaus) [reply core/camel-main/src/main/java/org/apache/camel/main/BaseMainSupport.java]: Ah yeah now i see it, yeah this is good as-is

**Pull request merged**